src/proto/data_logging.proto

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2020 Nick Mertin
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package bowler.data_logging;
+
+import "google/protobuf/timestamp.proto";
+
+option java_multiple_files = true;
+option java_package = "com.commonwealthrobotics.proto.data_logging";
+
+// Next: 1
+// The request message for listing sources
+message ListSourcesRequest {}
+
+// Next: 2
+// The response message for listing sources
+message ListSourcesResponse {
+    // The available sources
+    map<string, SourceMetadata> sources = 1;
+}
+
+// Next: 2
+// The metadata of an available source
+message SourceMetadata {
+    // The human-readable description of the source 
+    string description = 1;
+}
+
+// Next: 2
+// The request message for watching a source
+message WatchRequest {
+    // The ID of the source to watch
+    string source_id = 1;
```

Comment:
As before, I think we should merge source ID and name, so this can be `source_name`.